src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -8,42 +9,39 @@ namespace ePiggyWeb.DataManagement.Saving
 {
     public static class AlternativeSavingCalculator
     {
-        //This might not be 1M in the future, that why we are keeping this
-        private static decimal SavingRatio { get; } = 1M;
         private static decimal RegularSavingValue { get; } = 0.25M;
         private static decimal MaximalSavingValue { get; } = 0.5M;
         private static decimal MinimalSavingValue { get; } = 0.1M;
 
-        public static void GetSuggestedExpensesOffers(IEntryList entryList, IGoal goal, IList<ISavingSuggestion> entrySuggestions,
+        public static int GetSuggestedExpensesOffers(IEntryList entryList, IGoal goal, IList<ISavingSuggestion> entrySuggestions,
             List<SavingSuggestionByMonth> monthlySuggestions, decimal startingBalance, SavingType savingType = SavingType.Regular)
         {
             entrySuggestions ??= new List<ISavingSuggestion>();
 
             if (entryList is null)
             {
-                return;
+                return 0;
             }
 
             var enumCount = Enum.GetValues(typeof(Importance)).Length;
-
-            decimal[] arrayTemplate = new decimal[enumCount];
-            decimal[] sumsOfAmountByImportanceAdjusted = arrayTemplate;
-            decimal[] sumsOfAmountByImportanceDefault = arrayTemplate;
-            decimal[] averagesOfAmountByImportanceAdjusted = arrayTemplate;
-            decimal[] averagesOfAmountByImportanceDefault = arrayTemplate;
+ 
+            decimal[] sumsOfAmountByImportanceAdjusted = new decimal[enumCount];
+            decimal[] sumsOfAmountByImportanceDefault = new decimal[enumCount];
+            decimal[] averagesOfAmountByImportanceAdjusted = new decimal[enumCount];
+            decimal[] averagesOfAmountByImportanceDefault = new decimal[enumCount];
             int[] entryAmounts = new int[enumCount];
 
             for (var i = enumCount; i > (int)Importance.Necessary; i--)
             {
-                var expenses = entryList.GetBy((Importance)i - 1);
+                var expenses = entryList.GetBy((Importance)i);
                 var ratio = enumCount - i;
                 foreach (var entry in expenses)
                 {
                     var amountAfterSaving = savingType switch
                     {
-                        SavingType.Minimal => entry.Amount * SavingRatio * ratio * MinimalSavingValue,
-                        SavingType.Regular => entry.Amount * SavingRatio * ratio * RegularSavingValue,
-                        SavingType.Maximal => entry.Amount * SavingRatio * ratio * MaximalSavingValue,
+                        SavingType.Minimal => entry.Amount * ratio * MinimalSavingValue,
```

Comment:
Essentially the whole calculation works with two big cycles, I can't really imagine a way to split those up in a way that results in LESS confusing logic, because you'd have to trace what values and how are they worked with in different methods. I would consider it if you provided a better example of what could be done